Src/FluentAssertions/Primitives/StringAssertions.cs

```diff
@@ -1973,6 +1976,62 @@ public AndConstraint<TAssertions> NotBeLowerCased(string because = "", params ob
         return new AndConstraint<TAssertions>((TAssertions)this);
     }
 
+#if NET6_0_OR_GREATER
+    /// <summary>
+    /// Asserts that the string is a valid JSON document.
+    /// </summary>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public AndWhichConstraint<TAssertions, JsonDocument> BeValidJson(
+        string because = "", params object[] becauseArgs)
+    {
+        return BeValidJson(default, because, becauseArgs);
+    }
+
+    /// <summary>
+    /// Asserts that the string is a valid JSON document.
+    /// </summary>
+    /// <param name="options">
+    /// Options to control the reader behavior during parsing.
+    /// </param>
+    /// <param name="because">
+    /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+    /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+    /// </param>
+    /// <param name="becauseArgs">
+    /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+    /// </param>
+    public AndWhichConstraint<TAssertions, JsonDocument> BeValidJson(
+        JsonDocumentOptions options,
+        string because = "", params object[] becauseArgs)
+    {
+        JsonDocument json = null;
+
+        Execute.Assertion
+            .ForCondition(Subject is not null)
+            .BecauseOf(because, becauseArgs)
+            .FailWith("Expected {context:string} to be valid JSON{reason}, but found null.");
+
+        try
+        {
+            json = JsonDocument.Parse(Subject, options);
+        }
+        catch (JsonException jsonException)
+        {
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .FailWith("Expected {context:string} to be valid JSON{reason}, but parsing failed with {0}.", jsonException.Message);
+        }
+
+        return new AndWhichConstraint<TAssertions, JsonDocument>((TAssertions)this, json);
```

Comment:
This way of using JsonDocument seems to be against the documentation that states that it is important to dispose (or clone) JsonDocument instances. I'm not sure I fully understand the consequences they warn about here, but my suggestion would be to use the raw `Utf8JsonReader` for `BeValidJson` and not provide the document for further usage.

https://learn.microsoft.com/en-us/dotnet/standard/serialization/system-text-json/use-dom#jsondocument-is-idisposable

> JsonDocument builds an in-memory view of the data into a pooled buffer. Therefore the JsonDocument type implements IDisposable and needs to be used inside a using block.